src/js/fluxcontroller.js

```diff
@@ -860,8 +874,10 @@ define(function (require, exports, module) {
 
                         log.error("Action " + actionName + " failed:", message);
 
-                        // Reset all action modules on failure
-                        this._resetController(err);
+                        if (!action.action.noResetOnFailure) {
+                            // Reset all action modules on failure
+                            this._resetController(err);
+                        }
                         throw err;
```

Comment:
It's not obvious to me that you want to throw the err here even if `allowFailure` is set...